moduleroot/.github/workflows/acceptance.yml.erb

```diff
@@ -0,0 +1,54 @@
+<%- if Dir[File.join(@metadata[:workdir], 'spec', 'acceptance', '**', '*_spec.rb')].any? -%>
+name: Acceptance tests
+
+on:
+  - pull_request
+  - push
+
+jobs:
+  acceptance:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        setfile:
+          <%- require 'puppet_metadata' -%>
+          <%- metadata_file = File.join(@metadata[:workdir], 'metadata.json') -%>
+          <%- PuppetMetadata.read(metadata_file).beaker_setfiles(use_fqdn: true, pidfile_workaround: @configs['pidfile_workaround']) do |setfile| -%>
+          - <%= setfile %>
+          <%- end -%>
```

Comment:
again, code in a template, feels odd

Reply:
Yes, but the idea is to read `metadata.json` inside the module itself and generate setfiles based on the supported operating systems. This avoids the need to set it in three places.